File: javascript_to_golang/experiment11.go
package main

import "fmt"

func generateData() []string {
	number := "0821234567"
	var customers []string
	for i := 0; i < 100; i++ {
		var mobileNumber string
		if i < 10 {
			mobileNumber = number + "0" + fmt.Sprintf("%d", i)
		} else {
			mobileNumber = number + fmt.Sprintf("%d", i)
		}
		customers = append(customers, mobileNumber)
	}
	return customers
}

func sendPromoDiscount(array []string) {
	for i := 0; i < len(array); i++ {
		fmt.Printf("Sending Promo to %s\n", array[i])
	}
	for i := 0; i < 10; i++ {
		fmt.Printf("Sending Discount to Chosen Customer %d\n", i+1)
	}
}

func main() {
	customers := generateData()
	sendPromoDiscount(customers)
}

File: javascript_to_java/experiment-11.java
class experiment11 {
    public static String[] generateData() {
        String number = "0821234567";
        String[] customers = new String[100];
        String mobileNumber;

        for (int i = 0; i < 100; i++) {
            if (i < 10) {
                mobileNumber = number + "0" + i;
            } else {
                mobileNumber = number + i;
            }
            customers[i] = mobileNumber;
        }

        return customers;
    }

    public static void sendPromoDiscount(String[] array) {
        for (int i = 0; i < array.length; i++) {
            System.out.println("Sending Promo to " + array[i]);
        }
        for (int i = 0; i < 10; i++) {
            System.out.println("Sending Discount to Chosen Customer " + (i + 1));
        }
    }

    public static void main(String[] args) {
        String[] customers = generateData();
        sendPromoDiscount(customers);
    }
}
